src/test/java/com/example/expense/tracker/service/UserServiceTest.java

```diff
@@ -0,0 +1,127 @@
+package com.example.expense.tracker.service;
+
+import com.example.expense.tracker.repository.UserRepository;
+import com.example.expense.tracker.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserService userService;
+
+    @BeforeEach
+    public void setUp() {
+        // Reset mock before each test method
+        Mockito.reset(userRepository);
+    }
+
+    @Test
+    public void testAuthenticate_ValidCredentials_ReturnsTrue() {
+        // Arrange
+        String username = "testuser";
+        String password = "password";
+        // Mock UserRepository behavior
+        Mockito.when(userRepository.existsByUsernameAndPassword(username, password)).thenReturn(true);
+
+        // Act
+        boolean result = userService.authenticate(username, password);
+
+        // Assert
+        assertTrue(result);
+    }
+
+    @Test
+    public void testAuthenticate_InvalidCredentials_ReturnsFalse() {
+        // Arrange
+        String username = "testuser";
+        String password = "password";
+        // Mock UserRepository behavior
+        Mockito.when(userRepository.existsByUsernameAndPassword(username, password)).thenReturn(false);
+
+        // Act
+        boolean result = userService.authenticate(username, password);
+
+        // Assert
+        assertFalse(result);
+    }
+
+    @Test
+    public void testUsernameExists_ExistingUsername_ReturnsTrue() {
+        // Arrange
+        String existingUsername = "existingUser";
+        // Mock UserRepository behavior
+        Mockito.when(userRepository.existsByUsername(existingUsername)).thenReturn(true);
+
+        // Act
+        boolean result = userService.usernameExists(existingUsername);
+
+        // Assert
+        assertTrue(result);
+    }
+
+    @Test
+    public void testUsernameExists_NonExistingUsername_ReturnsFalse() {
+        // Arrange
+        String nonExistingUsername = "nonExistingUser";
```

Comment:
use RandomAlphabetic instead of constant value.  Add the dependency as well.

Ex:
import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;

String nonExistingUsername = randomAlphabetic(10) ;

Reply:
userServiveTest is not not neccessary so userServiveTest is removed